cvat/apps/engine/media_extractors.py

```diff
@@ -179,7 +183,8 @@ def _make_name():
 
 class ZipReader(ImageListReader):
     def __init__(self, source_path, step=1, start=0, stop=None):
-        self._zip_source = zipfile.ZipFile(source_path[0], mode='r')
+        self._dimension = DimensionType.DIM_2D
```

Comment:
Andrey and I asked you to reimplement the class. It is much better to create a class like LidarDataReader inherited from ZipReader and redefine all corresponding methods. Thus you will not need to have `if self._dimension == DimensionType.DIM_3D:` mostly in every method. Let me know if you disagree by a reason.

Reply:
As mentioned in the comments earlier, We distinguish the mime types only after we initialise the zipWriter class as initially we only enquire on what is the extension of the main file, which here is a zip (MediaFile). After the extraction along with picking the  names of possible image files, the actual zip is deleted by keeping file pointer open in memory. 

Upon deleting of zipExtractor object, the __del__ would close the open file pointer, which is being actively
utilized further in the process. Thus introduction of LidarData Class would disrupt the flow and make it required to 
create another zip file or abandon the deletion in the ZipExtractor. 
Also if the ` 2 if` are removed, then we need to introduce `if` in places such as FrameProvider, task.py and possibly when
import export is being carried based on dimension.
Kindly let us know your comments on the same.